File: app/src/main/java/com/arash/altafi/room/java/user_database.java
package com.arash.altafi.room.java;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(version = 1 , entities = user_model.class)
public abstract class user_database extends RoomDatabase {

    private static user_database database;

    public static user_database getDatabase(Context context) {
        if (database == null)
        {
            database = Room.databaseBuilder(context , user_database.class , "db_user").allowMainThreadQueries().build();
        }
        return database;
    }

    public abstract user_dao dao();

}

File: app/src/main/java/com/arash/altafi/room/kotlin/user_database.kt
package com.arash.altafi.room.kotlin

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(version = 1 , entities = arrayOf(user_model::class))
abstract class user_database:RoomDatabase() {

    companion object
    {
        var userDatabase:user_database ?= null
        fun getAppDataBase(context: Context) : user_database?
        {
            if (userDatabase == null)
            {
                synchronized(user_database::class)
                {
                    userDatabase = Room.databaseBuilder(context , user_database::class.java , "db_user_kotlin").allowMainThreadQueries().build()
                }
            }
            return userDatabase
        }
    }

    abstract fun UserDao() : user_dao

}